### PR TITLE
Aria scan fixes + optimize SliceDictionaryStreamReader

### PR DESCRIPTION
**Pull request by mbasmanova**
Fix queries with filters on multiple columns and output columns being a subset of input columns.

```
SELECT
   linenumber,
   orderkey
FROM lineitem
WHERE
   shipinstruct = 'TAKE BACK RETURN' AND orderkey < 10
```

TestAriaHiveDistributedQueries is passing.

### CODE REVIEW

File: presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDictionaryStreamReader.java [hunk 1]
```diff
@@ -206,6 +206,12 @@ private void setDictionaryBlockData(byte[] dictionaryData, int[] dictionaryOffse
             isNullVector[positionCount - 1] = true;
             dictionaryOffsets[positionCount] = dictionaryOffsets[positionCount - 1];
             dictionaryBlock = new VariableWidthBlock(positionCount, wrappedBuffer(dictionaryData), dictionaryOffsets, Optional.of(isNullVector));
+            if (filter != null && filter.isDeterministic()) {
+                if (filterResults == null || filterResults.length < positionCount) {
+                    filterResults = new byte[positionCount];
+                }
+                Arrays.fill(filterResults, (byte) 0);
```
Comment (elonazoulay) [inline presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDictionaryStreamReader.java]: No need to fill a newly created array. What do you think a out the following?
```
if (filterResults == null || filterResults.length < positionCount) {
    filterResults = new byte[positionCount]
}
else {
    Arrays.fill(filterResults, (byte) 0);
}
```

**Comment by mbasmanova** [comment]
@elonazoulay @tdcmeehan Elon, Tim, thanks for reviews. Comments addressed.

File: presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDictionaryStreamReader.java [hunk 2]
```diff
@@ -489,13 +500,26 @@ void reset()
...
             int dictionaryPosition = toIntExact(value);
+            if (!filter.isDeterministic()) {
+                return evaluateFilter(dictionaryPosition);
+            }
+
+            if (filterResults[dictionaryPosition] == FILTER_NOT_EVALUATED) {
+                filterResults[dictionaryPosition] = evaluateFilter(dictionaryPosition) ? FILTER_PASSED : FILTER_FAILED;
+            }
+
+            return filterResults[dictionaryPosition] == 1;
```
Comment (tdcmeehan) [inline presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDictionaryStreamReader.java]: Constant here too.

Comment (mbasmanova) [reply presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDictionaryStreamReader.java]: @tdcmeehan Thank you, Tim. Fixed.

**Comment by elonazoulay** [comment]
@tdcmeehan had some suggestions, once those are addressed LGTM.

**Pull request merged**